integration/integration_test.go

```diff
@@ -9198,3 +9199,94 @@ func testNegotiatedALPNProtocols(t *testing.T, suite *integrationTestSuite) {
 		})
 	}
 }
+
+func testForceListenerInTunnelMode(t *testing.T, suite *integrationTestSuite) {
+	// InsecureDevMode needed for IoT node handshake
+	lib.SetInsecureDevMode(true)
+	defer lib.SetInsecureDevMode(false)
```

Comment:
I'm always scared of global state manipulation in tests.

```suggestion
func testForceListenerInTunnelMode(t *testing.T, suite *integrationTestSuite) {
	// InsecureDevMode needed for IoT node handshake
	defer lib.SetInsecureDevMode(lib.IsInsecureDevMode())
	lib.SetInsecureDevMode(true)
	// fail the test if it's accidentally made parallel
	t.SetEnv("_testForceListenerInTunnelMode_SetInsecureDevMode", "1")
```